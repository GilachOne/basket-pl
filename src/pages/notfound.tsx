import React from 'react';
import Lottie from 'lottie-react';
import styled from '@emotion/styled';
import { useTranslation } from 'react-i18next';

import { animations } from '../__data__';

const TopWrapper = styled.div`
    display: flex;
    align-items: center;
    flex-direction: column;
    width: 100%;
    height: 100%;
`;

const AniWrapper = styled.div`
    top: 50%;
    left: 50%;
    position: fixed;
    transform: translate(-50%, -50%);
`;

export const NotFoundPage = () => {
    const { t } = useTranslation();

    return (
        <>
            <TopWrapper>
                <h1>{t('basket.notfoundpage.title')}</h1>
                <AniWrapper>
                    <Lottie animationData={animations.oopss} />
                </AniWrapper>
            </TopWrapper>
        </>
    );
};
