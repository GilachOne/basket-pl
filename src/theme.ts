export const theme = {
    colors: {
        brand: {
            light2: '#08DA36',
            light: '#74D14C',
            main: '#74D14C',
            dark: '#54AB26',
        },
        background: {
            light: '#FFFFFF',
			main: '#008060',
        },
        accent: {
            light6: '#CCCCCC',
            light5: '#F6EAFF',
            light4: '#d2d8e0',
            light3: '#FAF9F8',
            light2: '#D2D5D9',
            light: '#F6EAFF',
            main: '#9828F0',
            dark: '#2C107A',
        },
        text: {
            light3: '#A68FE5',
            light2: '#42474C',
            light: '#6B7177',
            main: '#FFFFFF',
            dark: '#000000',
        },
        bg: {
            main: '#000000',
        },
        error: {
            main: '#FF0000',
        },
    },
};
