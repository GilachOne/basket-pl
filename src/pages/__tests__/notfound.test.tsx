import React from 'react';
import { describe, it, expect } from '@jest/globals';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { BrowserRouter } from 'react-router-dom';

import { theme } from '../../theme';
import { NotFoundPage } from '../notfound';

// eslint-disable-next-line react/display-name
jest.mock('lottie-react', () => () => <h1>Страница не найдена</h1>);

describe('Not Found Page', () => {

    it('Not Found Test', () => {
        const { container } = render(
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <NotFoundPage />
                    </BrowserRouter>
                </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });
});