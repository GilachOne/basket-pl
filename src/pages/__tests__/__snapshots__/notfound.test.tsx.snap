// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page Not Found Test 1`] = `
<div>
  <div
    class="css-1d342t8"
  >
    <h1>
      Страница не найдена
    </h1>
    <div
      class="css-8428uh"
    >
      <h1>
        Страница не найдена
      </h1>
    </div>
  </div>
</div>
`;
